### PR TITLE
Requirements API

### PR DESCRIPTION
**Pull request by oxisto**
This adds new endpoints to manage the selection of requirements / criteria of a cloud service in the orchestrator.

### CODE REVIEW

File: service/orchestrator/requirements.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"clouditor.io/clouditor/api/orchestrator"
+)
+
+// LoadRequirements loads requirements definitions from a JSON file.
+func LoadRequirements(file string) (requirements []*orchestrator.Requirement, err error) {
```
Comment (lebogg) [inline service/orchestrator/requirements.go:37]: Maybe use this function in `CreateDefaultTargetCloudService` to add default requirements to the default service?

**Review by lebogg — COMMENTED** [review]
Almost good to go :)

File: cli/session.go [hunk 2]
```diff
@@ -326,6 +318,34 @@
 	return metrics
 }
 
+// TODO(oxisto): This could be an interesting use case for 1.18 Go generics
+func getRequirements(_ string) []string {
```
[the remark this answers is not in the document]
Comment (oxisto) [reply cli/session.go:322]: Yes, future-ready.

File: service/orchestrator/requirements.go (hunk 1 above)
Comment (oxisto) [reply service/orchestrator/requirements.go:37]: Not sure what you mean

Comment (lebogg) [reply service/orchestrator/requirements.go:37]: in `service/orchestrator/cloud_service.go:159`, the function `CreateDefaultTargetCloudService` creates a default target service. Currently with no requirements associated.
My question is, if it makes sense to load some default requirements into this `defaultTargetService`?

File: cli/session.go [hunk 3]
```diff
@@ -247,27 +247,19 @@
 	return []string{}, cobra.ShellCompDirectiveNoFileComp
 }
 
-func ValidArgsGetTools(_ *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
-	if len(args) != 0 {
-		return nil, cobra.ShellCompDirectiveNoFileComp
-	}
-
+func ValidArgsGetTools(_ *cobra.Command, _ []string, toComplete string) ([]string, cobra.ShellCompDirective) {
 	return getTools(toComplete), cobra.ShellCompDirectiveNoFileComp
```
[the remark this answers is not in the document]
Comment (oxisto) [reply cli/session.go:251]: I have no idea why this was there in the first place. It should not be :)

File: service/orchestrator/requirements.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+		return nil, fmt.Errorf("error in JSON marshal: %w", err)
+	}
+
+	return requirements, nil
+}
+
+func (svc *Service) ListRequirements(_ context.Context, _ *orchestrator.ListRequirementsRequest) (response *orchestrator.ListRequirementsResponse, err error) {
+	return &orchestrator.ListRequirementsResponse{
+		Requirements: svc.requirements,
+	}, nil
```
[the remark this answers is not in the document]
Comment (oxisto) [reply service/orchestrator/requirements.go:58]: ?

File: service/orchestrator/orchestrator.go [hunk 5]
```diff
@@ -72,6 +73,8 @@
 	storage persistence.Storage
 
 	metricsFile string
+
+	requirements []*orchestrator.Requirement
```
[the remark this answers is not in the document]
Comment (oxisto) [reply service/orchestrator/orchestrator.go:77]: the array is easier to load. for now it is sufficient to just have list requirements,

File: cli/commands/requirement/requirement.go [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+package requirement
+
+import (
+	"clouditor.io/clouditor/cli/commands/service/orchestrator"
+	"github.com/spf13/cobra"
+)
+
+// NewListRequirementsCommand returns a cobra command for the `list` subcommand
+func NewListRequirementsCommand() *cobra.Command {
+	// Use Orchestrator's function for listing assessment results
```
Comment (lebogg) [inline cli/commands/requirement/requirement.go:35]:
```suggestion
	// Use Orchestrator's function for listing requirements
```

File: cli/commands/requirement/requirement.go [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+	cmd := orchestrator.NewListRequirementsCommand()
+
+	// Change use for better readability
+	cmd.Use = "list"
+	return cmd
+}
+
+// NewRequirementCommand returns a cobra command for `metric` subcommands
```
Comment (lebogg) [inline cli/commands/requirement/requirement.go:43]:
```suggestion
// NewRequirementCommand returns a cobra command for `requirement` subcommands
```

File: cli/commands/service/orchestrator/orchestrator.go [hunk 8]
```diff
@@ -68,6 +68,38 @@ func NewListAssessmentResultsCommand() *cobra.Command {
...
+		},
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			return []string{}, cobra.ShellCompDirectiveNoFileComp
+		},
+	}
+
+	return cmd
+}
+
 // NewOrchestratorCommand returns a cobra command for `assessment` subcommands
```
Comment (lebogg) [inline cli/commands/service/orchestrator/orchestrator.go:103]:
```suggestion
// NewOrchestratorCommand returns a cobra command for `orchestrator` subcommands
```

**Review by lebogg — APPROVED** [review]
Only minor stuff.

File: service/orchestrator/requirements.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (oxisto) [reply service/orchestrator/requirements.go:62]: Yes. I was using `srv` for server and `svc` for service.

**Pull request merged**